### ISSUE TITLE
emqx broker does not support large file publish

### ISSUE THREAD

**Issue by moluzhui** [emqx/emqx#3102]
#### Environment
ubuntu18.04

emqx:latest

#### Description

*A description of the issue*
I need to publish a big file of about 20M, I have setting max_packet_size = 50M, but it is still no transmission

Part of the information is as follows:
The file size is set by MQTT's `enable_logger`
```
DEBUG:mqtt_client:Sending PUBLISH (d0, q2, r0, m2), 'b'detector_1'', ... (18662731 bytes)
```

emqx.conf
```
mqtt.max_packet_size = 50MB
```

logs error
```
2019-12-12 13:02:01.344 [warning] [SYSMON] long_schedule warning: pid = <0.1980.0>, info: [{timeout,258},
                                                {in,
                                                 {emqx_channel,
                                                  process_incoming,3}},
                                                {out,
                                                 {emqx_channel,next_events,
                                                  1}}]
[{initial_call,{proc_lib,init_p,5}},
 {current_function,{gen_statem,loop_receive,3}},
 {registered_name,[]},
 {status,running},
 {message_queue_len,0},
 {group_leader,<0.1353.0>},
 {priority,normal},
 {trap_exit,true},
 {reductions,515351},
 {last_calls,false},
 {catchlevel,1},
 {trace,0},
 {suspending,[]},
 {sequential_trace_token,[]},
 {error_handler,error_handler},
 {memory,39728},
 {total_heap_size,4805},
 {heap_size,610},
 {stack_size,10},
 {min_heap_size,233}]
```

What caused the size setting to fail?

**Comment by tigercl** [comment]
@moluzhui The warning means that your CPU has reached a bottleneck, and the warning itself does not cause the message to fail. I was able to transmit 20MB of messages on my own machine, I suggest you upgrade the machine configuration.

**Comment by moluzhui** [comment]
The CPU information is as follows, and I don't think this will be a bottleneck
```
架构：           x86_64
CPU 运行模式：   32-bit, 64-bit
字节序：         Little Endian
CPU:             40
在线 CPU 列表：  0-39
每个核的线程数： 2
每个座的核数：   10
座：             2
NUMA 节点：      2
厂商 ID：        GenuineIntel
CPU 系列：       6
型号：           79
型号名称：       Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz
步进：           1
CPU MHz：        1197.352
CPU 最大 MHz：   3400.0000
CPU 最小 MHz：   1200.0000
BogoMIPS：       4789.05
虚拟化：         VT-x
L1d 缓存：       32K
L1i 缓存：       32K
L2 缓存：        256K
L3 缓存：        25600K
NUMA 节点0 CPU： 0-9,20-29
NUMA 节点1 CPU： 10-19,30-39
```
There is a problem if I start emqx docker.Can you tell me your `startup command` and some necessary configuration information, such as `emqx.conf` and the like.

**Comment by tigercl** [comment]
@moluzhui How many clients are publishing large messages at the same time?

**Comment by moluzhui** [comment]
@tigercl When we tested it, only one client sent message.  After testing, it was found that the maximum bytes emqx can send is `net.core.wmem_default`. `net.core.wmem_max` is set but invalid

**Comment by tigercl** [comment]
@moluzhui Did you restart emqx after you updated these system parameters?

**Comment by moluzhui** [comment]
@tigercl Of course. It's restarted every time. 

Can you provide the `docker command` that you successfully tested and the relevant `docker and system environment` or if you can provide `advice` on how to implement the `max_packet` setting?

**Comment by martinrdrz** [comment]
Good afternoon.
I have the EMQX message broker installed and I have modified the parameter "mqtt.max_packet_size = 128MB", but I cannot send files larger than 20 MBytes, and even messages of 10MBytes and 20MBytes take up to 90 seconds to be transmitted, which it is a long time compared to "Broker Mosquitto" that takes 20 seconds that same file size.
Do you know why it takes so long to send the 10/20 Mbyte files and why can't I send larger files?
Thank you very much.

**Comment by k32** [comment]
Hello,

This is a duplicate of https://github.com/emqx/emqx/issues/4787
This issue has been fixed in 4.3.1
